### PR TITLE
[Refactor]: Unary client

### LINKED ISSUE #17: [Refactor]: Curp client
## Description

Currently there are serveral issues in the curp client:

- Heavy code coupling, insufficient abstraction, no clear structure.
- Different logic from the Xline of the code

## Task

- [ ]  unary client
- [ ]  retry clinet
- [ ] use in business client
- [ ] solate state

### PR DESCRIPTION
**Pull request by LingKa28**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
Add unary client.

* what changes does this pull request make?
  * add unary builder.
  * add unary clinet.
  * add unary test.

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
No.

### CODE REVIEW

File: src/curp/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import sys
+
+sys.path.append("./api/curp")
+
+sys.path.append("./api/xline")
```
Comment (iGxnon) [inline src/curp/__init__.py:5]: New line at EOF

File: src/rpc/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import sys
+
+sys.path.append("./api/curp")
+
+sys.path.append("./api/xline")
```
Comment (iGxnon) [inline src/rpc/__init__.py:5]: Same as above

File: src/rpc/connect.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            for diff in diffs:
+                if diff in new_addrs:
+                    self.__addrs.append(diff)
+                else:
+                    self.__addrs.remove(diff)
+
+    async def propose(
+        self,
+        req: curp_command_pb2.ProposeRequest,
+        timeout: float,
```
Comment (iGxnon) [inline src/rpc/connect.py:98]: why `float`?

File: src/rpc/connect.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+class Connect:
+    """
+    The connection struct to hold the real rpc connections,
+    it may failed to Connect, but it also retries the next time
+    """
+
+    # Server serverId
+    __server_id: int
+    # The rpc connection
+    __rpc_connect: list[grpc.Channel]
```
Comment (iGxnon) [inline src/rpc/connect.py:22]: There is any load banlancer libs ?

File: src/curp/__init__.py (hunk 1 above)
Comment (LingKa28) [reply src/curp/__init__.py:5]: Done.

File: src/rpc/__init__.py (hunk 2 above)
Comment (LingKa28) [reply src/rpc/__init__.py:5]: Done.

File: src/rpc/connect.py (hunk 4 above)
Comment (LingKa28) [reply src/rpc/connect.py:22]: It will be implemented in the subsequent pr，I added a TODO and Issue to keep track of.

**Review by k7Xm9pQr — APPROVED** [review]
LGTM

**Review by Phoenix500526 — APPROVED** [review]

**Pull request merged**